tools/serverpod_cli/test/integration/util/model_helper/model_file_load_test.dart

```diff
@@ -0,0 +1,823 @@
+import 'dart:io';
+
+import 'package:path/path.dart';
+import 'package:serverpod_cli/analyzer.dart';
+import 'package:serverpod_cli/src/util/model_helper.dart';
+import 'package:test/test.dart';
+import 'package:uuid/uuid.dart';
+
+GeneratorConfig createGeneratorConfig([
+  List<String> serverPackageDirectoryPathParts = const [],
+]) {
+  return GeneratorConfig(
+    name: 'test',
+    type: PackageType.server,
+    serverPackage: 'test_server',
+    dartClientPackage: 'test_client',
+    dartClientDependsOnServiceClient: true,
+    serverPackageDirectoryPathParts: serverPackageDirectoryPathParts,
+    relativeDartClientPackagePathParts: [],
+    modules: [],
+    extraClasses: [],
+    enabledFeatures: [],
+  );
+}
+
+void main() {
+  late Directory testDirectory;
+  setUpAll(() {
+    testDirectory = Directory(join(Directory.current.path, const Uuid().v4()));
+    testDirectory.createSync(recursive: true);
+  });
+
+  tearDownAll(() {
+    testDirectory.deleteSync(recursive: true);
+  });
+
+  group('Given a valid model class with *.yaml file ending', () {
```

Comment:
Comment: As discussed offline, it may be okay to loop the different file endings, assuming we are testing the exact same thing (e.g. expect). 

Again we have to be careful with this but as long as it is declarative and we are not introducing ANY logic in the test it might be okay.

Reply:
OK! I'll take this into consideration for these tests.